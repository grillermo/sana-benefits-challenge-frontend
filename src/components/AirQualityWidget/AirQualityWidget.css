.airQualityWidget {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 200px;
  padding: 20px;
  border-radius: 15px;
  color: white;
}

.airQualityWidget.ok {
  background-color: green;
}

.airQualityWidget.disabled {
  background-color: gray;
}

.airQualityWidget.danger {
  background-color: red;
}
