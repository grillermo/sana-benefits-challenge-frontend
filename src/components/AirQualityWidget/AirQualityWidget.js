import { isNumber } from 'lodash'

import './AirQualityWidget.css';

export default function AirQualityWidget(props) {
  const { currentLevel, threshold } = props

  const valuesPresent = isNumber(currentLevel) && isNumber(threshold)

  let className = 'disabled'

  if(valuesPresent) {
    if(currentLevel > threshold) {
      className = 'ok'
    } else if (currentLevel <= threshold) {
      className = 'danger'
    }
  }

  let levelFriendlyName

  switch(className) {
    case 'ok':
      levelFriendlyName = 'All good'
      break
    case 'danger':
      levelFriendlyName = 'Dangerous'
      break
    default:
      levelFriendlyName = 'loading...'
      break
  }

  return(
    <div className={ 'airQualityWidget '+className }>
      <h1>
        {valuesPresent ? currentLevel : 'loading...'}
      </h1>
      <strong>
        current level
      </strong>
      <p>
        {levelFriendlyName}
      </p>
    </div>
  )
}
