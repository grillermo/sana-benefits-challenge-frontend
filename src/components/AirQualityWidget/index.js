import AirQualityWidget from './AirQualityWidget'
export default AirQualityWidget
