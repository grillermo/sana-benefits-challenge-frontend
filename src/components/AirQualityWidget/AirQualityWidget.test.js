import { shallow } from 'enzyme';

import AirQualityWidget from './AirQualityWidget'

describe('AirQualityWidget ', () => {

  describe('with empty values as props', () => {
    test('renders successfully', () => {
      const airQualityWidget = shallow(<AirQualityWidget currentLevel={null} threshold={''} />)

      expect(airQualityWidget).toMatchSnapshot()
    })
  })

  describe('with current level less than threshold', () => {
    test('renders successfully', () => {
      const airQualityWidget = shallow(<AirQualityWidget currentLevel={70} threshold={80} />)

      expect(airQualityWidget).toMatchSnapshot()
    })
  })

  describe('with current level more than threshold', () => {
    test('renders successfully', () => {
      const airQualityWidget = shallow(<AirQualityWidget currentLevel={100} threshold={80} />)

      expect(airQualityWidget).toMatchSnapshot()
    })
  })
})

