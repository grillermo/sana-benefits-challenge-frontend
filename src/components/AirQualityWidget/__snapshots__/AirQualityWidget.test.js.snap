// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirQualityWidget  with current level less than threshold renders successfully 1`] = `
<div
  className="airQualityWidget danger"
>
  <h1>
    70
  </h1>
  <strong>
    current level
  </strong>
  <p>
    Dangerous
  </p>
</div>
`;

exports[`AirQualityWidget  with current level more than threshold renders successfully 1`] = `
<div
  className="airQualityWidget ok"
>
  <h1>
    100
  </h1>
  <strong>
    current level
  </strong>
  <p>
    All good
  </p>
</div>
`;

exports[`AirQualityWidget  with empty values as props renders successfully 1`] = `
<div
  className="airQualityWidget disabled"
>
  <h1>
    loading...
  </h1>
  <strong>
    current level
  </strong>
  <p>
    loading...
  </p>
</div>
`;
